### ISSUE TITLE
Multi level nested properties not getting patched

### ISSUE THREAD

**Issue by brikesh987** [OData/WebApi#2238]
*Short summary (3-5 sentences) describing the issue.*
I've an entity with following nested navigational properties. I want to patch the person object with change to the address fields.
The delta on the server always have null for Address field

**PATCH ~/Persons/{key}**
`
Person {
    "Location":{
             "Address":{
                  "City": "Some city",
                  "State": "Some State"
              },
              "Latitude" : 1,
              "Longitude": 2
     }
}
`

I read that multi level patch is not supported by design. If that's the case, whats the best way to patch the address property for the person object? I tried following route but location being a complex type with no Id, didn't work

**PATCH ~/Persons/{key}/Location**
`"Location":{
             "Address":{
                  "City": "Some city",
                  "State": "Some State"
              },
              "Latitude" : 1,
              "Longitude": 2
     }`

For now I've created a bounded action which maps to following url and it works.
**Post ~/Persons/{key}/UpdateLocation**
`"Location":{
             "Address":{
                  "City": "Some city",
                  "State": "Some State"
              },
              "Latitude" : 1,
              "Longitude": 2
     }`

 However we want to have the url for the action as. But this doesn't work either. Please suggest the best way to patch multi level nested navigational property if it doesn't work as part of patch of the root entity.
**Post ~/Persons/{key}/Location**

### Assemblies affected

*Which assemblies and versions are known to be affected e.g. OData WebApi lib 6.1.0

### Reproduce steps

*The simplest set of steps to reproduce the issue. If possible, reference a commit that demonstrates the issue.*

### Expected result

*What would happen if there wasn't a bug.*

### Actual result

*What is actually happening.*

### Additional detail

*Optional, details of the root cause if known. Delete this section if you have no additional details to add.*

**Comment by xuzhg** [comment]
@brikesh987 What version do you use? 6.1 or 7.x? 

In 7.x, 

The follow scenarios should work ( if Location is complex type property)

1)

```
PATCH ~/Persons/{key}
{ "Location":{ "Address":{ "City": "Some city", "State": "Some State" }, "Latitude" : 1, "Longitude": 2 } }
```

2)

```
 PATCH ~/Persons/{key}/Location
{ "Address":{ "City": "Some city", "State": "Some State" }, "Latitude" : 1, "Longitude": 2 }
```

Please let us know the result?

**Comment by thisjgu** [comment]
@brikesh987 @xuzhg could you please share the reference regarding why "multi level patch is not supported by design."?

**Comment by KenitoInc** [comment]
Fixed by #2246
